pcdsdevices/gasdet.py

```diff
@@ -0,0 +1,345 @@
+"""
+Ophyd devices for the acqiris digitizer systems
+"""
+import logging
+from ophyd import EpicsSignal, EpicsSignalRO, Device
+from ophyd import Component as Cpt
+from ophyd import FormattedComponent as FCpt
+from .interface import BaseInterface
+
+logger = logging.getLogger(__name__)
+
+
+class acqiris_channel(Device):
+    """
+    Class to define a single acqiris channel.
+    """
+    waveform = Cpt(
+        EpicsSignalRO,
+        ':Data',
+        kind='hinted'
+    )
+    v_max = Cpt(
+        EpicsSignal,
+        ':CMaxVert',
+        kind='normal'
+    )
+    v_cent = Cpt(
+        EpicsSignal,
+        ':CCenter',
+        kind='normal'
+    )
+    v_in = Cpt(
+        EpicsSignal,
+        ':CMinVert',
+        kind='normal'
+    )
+    v_scale = Cpt(
+        EpicsSignal,
+        ':CVPerDiv',
+        kind='normal'
+    )
+    bandwidth = Cpt(
+        EpicsSignal,
+        ':CBandwidth',
+        kind='config',
+        string=True
+    )
+    full_scale = Cpt(
+        EpicsSignal,
+        ':CFullScale',
+        kind='config'
+    )
+    coupling = Cpt(
+        EpicsSignal,
+        ':CCoupling',
+        kind='config',
+        string=True
+    )
+    offset = Cpt(
+        EpicsSignal,
+        ':COffset',
+        kind='config'
+    )
+    trig_coupling = Cpt(
+        EpicsSignal,
+        'CTrigCoupling',
+        kind='config',
+        string=True
+    )
+    trig_slope = Cpt(
+        EpicsSignal,
+        'CTrigSlope',
+        kind='config',
+        string=True
+    )
+    trig_level = Cpt(
+        EpicsSignal,
+        ':CTrigLevel1',
+        kind='config'
+    )
+
+
+class acqiris(Device, BaseInterface):
+    """
+    Class to define an acqiris digitizer board as constructed by the
+    'GasDetDAQ' EPICS IOC. This device is used to access specific
+    acqiris channels as subcomponents.
+
+    Parameters
+    ----------
+    prefix : ``str``
+        The PV base of the digitizer.
+    platform : ``str``
+        This a legacy PV segment found in the GasDetDAQ IOC.
+        The default '202' applies to the HXR gas detector system.
+    module : ``str``
+        3-digit number representing the digitizer board,
+        e.g. '240', '360'.
+
+    """
+    ch1 = FCpt(
+        acqiris_channel,
+        '{self.prefix}:{self._mod_pref}1',
```

Comment:
These `self.` things are no longer needed in the newest ophyd, you can just do '{prefix}:{_mod_pref}1'